forem/forem
# [EOY2020 Issue] Default new forems to Email Authentication only

## What type of PR is this? (check all applicable)


## Description
This PR ensures that new Forems will default to Email Authentication only. The PR also ensures that an Authentication Provider is NOT enabled if either their respective Key or Secret is missing.

closes https://github.com/forem/InternalProjectPlanning/issues/336

## Related Tickets & Documents
None

## QA Instructions, Screenshots, Recordings

1. Navigate to `localhost:xxxx/admin/config` as a user with `:super_admin` role
2. Under the `Authentication` section, enable some Auth Providers, but ensure their key and secret are blank. You should see a warning banner about the empty Key and Secret fields. Update Site Configuration. You should find that the Auth Providers you updated were NOT enabled.
3. Repeat #2 but with random strings in the Auth Providers' Key and Secret fields. Update Site Configuration. You should find that the Auth Providers you updated ARE enabled. The warning banner should also be gone.

#### Demo Video:
https://www.loom.com/share/1b9d5a60946b4f469d9781e8f7de344b

### UI accessibility concerns?
None.

## Added tests?

- [X] Yes, all associated tests were updated
      have not been included_

## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?
None

## [optional] What gif best describes this PR or how it makes you feel?

![Fish Mooney issuing threats](https://media.giphy.com/media/GFLfcgMzh1BYY/giphy.gif)
